File: BubblerAndroid/app/src/main/java/com/bubbler/android/ui/components/BubblerLogo.kt
package com.bubbler.android.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.offset
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.blur
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import com.bubbler.android.app.theme.BubblerTheme

/** Bubble-cluster brand mark — mirrors Swift `BubblerLogoView`. */
@Composable
fun BubblerLogo(
    modifier: Modifier = Modifier,
    size: Dp = 180.dp,
) {
    val scale = size / 180.dp
    Box(
        modifier = modifier.size(size),
        contentAlignment = Alignment.Center,
    ) {
        Box(
            modifier = Modifier
                .size(size)
                .clip(CircleShape)
                .background(
                    Brush.radialGradient(
                        colors = listOf(
                            Color.Cyan.copy(alpha = 0.25f),
                            Color(0xFF2196F3).copy(alpha = 0.15f),
                            Color(0xFF2196F3).copy(alpha = 0.05f),
                            Color.Transparent,
                        ),
                    ),
                ),
        )

        LogoBubble(x = (-25).dp * scale, y = (-30).dp * scale, size = 38.dp * scale, opacity = 0.9f)
        LogoBubble(x = 15.dp * scale, y = (-45).dp * scale, size = 22.dp * scale, opacity = 0.7f)
        LogoBubble(x = 35.dp * scale, y = (-10).dp * scale, size = 26.dp * scale, opacity = 0.75f)
        LogoBubble(x = (-35).dp * scale, y = 15.dp * scale, size = 30.dp * scale, opacity = 0.8f)
        LogoBubble(x = 10.dp * scale, y = 25.dp * scale, size = 40.dp * scale, opacity = 0.85f)
        LogoBubble(x = 45.dp * scale, y = 35.dp * scale, size = 20.dp * scale, opacity = 0.65f)
        LogoBubble(x = (-10).dp * scale, y = 55.dp * scale, size = 24.dp * scale, opacity = 0.7f)
        LogoBubble(
            x = 5.dp * scale,
            y = 5.dp * scale,
            size = 18.dp * scale,
            opacity = 1.0f,
            highlight = true,
        )
    }
}

@Composable
private fun LogoBubble(
    x: Dp,
    y: Dp,
    size: Dp,
    opacity: Float,
    highlight: Boolean = false,
) {
    Box(
        modifier = Modifier
            .offset(x = x, y = y)
            .size(size)
            .then(
                if (highlight) {
                    Modifier.shadow(
                        elevation = 6.dp,
                        shape = CircleShape,
                        ambientColor = Color.White.copy(alpha = 0.3f),
                        spotColor = Color.White.copy(alpha = 0.3f),
                    )
                } else {
                    Modifier.shadow(
                        elevation = 4.dp,
                        shape = CircleShape,
                        ambientColor = Color(0xFF2196F3).copy(alpha = 0.15f),
                        spotColor = Color(0xFF2196F3).copy(alpha = 0.15f),
                    )
                },
            )
            .clip(CircleShape)
            .background(
                Brush.linearGradient(
                    colors = listOf(
                        Color.White.copy(alpha = opacity),
                        Color.Cyan.copy(alpha = opacity * 0.6f),
                        Color(0xFF2196F3).copy(alpha = opacity * 0.8f),
                    ),
                ),
            )
            .border(
                width = 1.dp,
                color = Color.White.copy(alpha = 0.80f),
                shape = CircleShape,
            )
            .blur(0.2.dp),
    )
}

@Preview(showBackground = true, backgroundColor = 0xFF0D47A1)
@Composable
private fun BubblerLogoPreview() {
    BubblerTheme {
        BubblerLogo()
    }
}

File: bubbler_app/lib/shared/widgets/bubbler_logo.dart
import 'package:flutter/material.dart';

import '../../app/theme.dart';

/// Bubble cluster logo matching Swift `BubblerLogoView`.
class BubblerLogo extends StatelessWidget {
  const BubblerLogo({super.key, this.size = 180});

  /// Outer frame size. Bubble offsets/diameters scale from a 180pt design.
  final double size;

  @override
  Widget build(BuildContext context) {
    final scale = size / 180;

    return SizedBox(
      width: size,
      height: size,
      child: Stack(
        alignment: Alignment.center,
        clipBehavior: Clip.none,
        children: [
          Container(
            width: size,
            height: size,
            decoration: BoxDecoration(
              shape: BoxShape.circle,
              gradient: RadialGradient(
                colors: [
                  BubblerTheme.cyan.withValues(alpha: 0.25),
                  BubblerTheme.blue.withValues(alpha: 0.15),
                  BubblerTheme.blue.withValues(alpha: 0.05),
                  Colors.transparent,
                ],
                stops: const [0.0, 0.35, 0.65, 1.0],
              ),
            ),
          ),
          _Bubble(x: -25, y: -30, diameter: 38, opacity: 0.9, scale: scale),
          _Bubble(x: 15, y: -45, diameter: 22, opacity: 0.7, scale: scale),
          _Bubble(x: 35, y: -10, diameter: 26, opacity: 0.75, scale: scale),
          _Bubble(x: -35, y: 15, diameter: 30, opacity: 0.8, scale: scale),
          _Bubble(x: 10, y: 25, diameter: 40, opacity: 0.85, scale: scale),
          _Bubble(x: 45, y: 35, diameter: 20, opacity: 0.65, scale: scale),
          _Bubble(x: -10, y: 55, diameter: 24, opacity: 0.7, scale: scale),
          _Bubble(
            x: 5,
            y: 5,
            diameter: 18,
            opacity: 1.0,
            scale: scale,
            highlight: true,
          ),
        ],
      ),
    );
  }
}

class _Bubble extends StatelessWidget {
  const _Bubble({
    required this.x,
    required this.y,
    required this.diameter,
    required this.opacity,
    required this.scale,
    this.highlight = false,
  });

  final double x;
  final double y;
  final double diameter;
  final double opacity;
  final double scale;
  final bool highlight;

  @override
  Widget build(BuildContext context) {
    final size = diameter * scale;

    return Transform.translate(
      offset: Offset(x * scale, y * scale),
      child: Container(
        width: size,
        height: size,
        decoration: BoxDecoration(
          shape: BoxShape.circle,
          gradient: LinearGradient(
            begin: Alignment.topLeft,
            end: Alignment.bottomRight,
            colors: [
              Colors.white.withValues(alpha: opacity),
              BubblerTheme.cyan.withValues(alpha: opacity * 0.6),
              BubblerTheme.blue.withValues(alpha: opacity * 0.8),
            ],
          ),
          border: Border.all(
            color: Colors.white.withValues(alpha: 0.8),
          ),
          boxShadow: [
            BoxShadow(
              color: BubblerTheme.blue.withValues(alpha: 0.15),
              blurRadius: 4 * scale,
            ),
            if (highlight)
              BoxShadow(
                color: Colors.white.withValues(alpha: 0.3),
                blurRadius: 6 * scale,
              ),
          ],
        ),
      ),
    );
  }
}
